packages/ton-client/src/shaders/sample.fs

```diff
@@ -1,65 +1,69 @@
 #version 300 es
 
-precision highp float; 
+precision highp float;
 
 uniform float sqrtBufferSize;
 uniform float oneByBufferSize;
 
-uniform float startAngle; 
+uniform float startAngle;
 uniform float endAngle;
 
-uniform float planeStartAngle; 
+uniform float planeStartAngle;
 uniform float planeEndAngle;
 
 uniform float radius;
 
-in vec2 uv; 
+uniform vec3 touchManipulationState;
 
-out vec4 rgba; 
+in vec2 uv;
+
+out vec4 rgba;
 
 // forward declaration for the imported sdf
-float sdf(vec3); 
+float sdf(vec3, vec3);
 
 void main() {
-  vec4 samples; 
+  vec4 samples;
   for(int s = 0; s < 4; s++) {
 
     float ix = (uv.x - 0.5) * 4. + float(s);
     float i = ix + (uv.y - 0.5) * sqrtBufferSize;
 
-    // plane 
+    // plane
     // the plane rotates every bar
     // the the axis of plane rotation rotates every 4 bars
     float planeAngle = mix(planeStartAngle, planeEndAngle, i * oneByBufferSize);
     vec3 planeX = vec3(
-      cos(planeAngle), 
-      sin(planeAngle), 
+      cos(planeAngle),
+      sin(planeAngle),
       0
     );
     vec3 planeZ = vec3(0, 0, 1);
 
+    vec3 scale = vec3(0.75 + 0.15 * touchManipulationState.x);
```

Comment:
try to move any calculations like these into the CPU thread. 
We don't want to do calculations like `0.75 + 0.15 * x` millions of times in shader invocations when it's sufficient to do them once on the CPU before passing them as uniforms.

Reply:
yes, I agree, thanks